src/CoreBundle/DependencyInjection/Compiler/FormCompilerPass.php

```diff
@@ -0,0 +1,158 @@
+                true,
+                'The "%alias_id%" service is deprecated since sonata-project/core-bundle 3.19 and will be removed in 4.0. Use "sonata.form.type.datetime_range" instead.'
+            );
+
+        $container
+            ->setAlias('sonata.core.form.type.date_picker', 'sonata.form.type.date_picker')
+            ->setPublic(true)
+            ->setDeprecated(
+                true,
+                'The "%alias_id%" service is deprecated since sonata-project/core-bundle 3.19 and will be removed in 4.0. Use "sonata.form.type.date_picker" instead.'
+            );
+
+        $container
+            ->setAlias('sonata.core.form.type.datetime_picker', 'sonata.form.type.datetime_picker')
+            ->setPublic(true)
+            ->setDeprecated(
+                true,
+                'The "%alias_id%" service is deprecated since sonata-project/core-bundle 3.19 and will be removed in 4.0. Use "sonata.form.type.datetime_picker" instead.'
+            );
+
+        $container
+            ->setAlias('sonata.core.form.type.date_range_picker', 'sonata.form.type.date_range_picker')
+            ->setPublic(true)
+            ->setDeprecated(
+                true,
+                'The "%alias_id%" service is deprecated since sonata-project/core-bundle 3.19 and will be removed in 4.0. Use "sonata.form.type.date_range_picker" instead.'
+            );
+
+        $container
+            ->setAlias('sonata.core.form.type.datetime_range_picker', 'sonata.form.type.datetime_range_picker')
+            ->setPublic(true)
+            ->setDeprecated(
+                true,
+                'The "%alias_id%" service is deprecated since sonata-project/core-bundle 3.19 and will be removed in 4.0. Use "sonata.form.type.datetime_range_picker" instead.'
+            );
+
+        $container
+            ->setAlias('sonata.core.form.type.equal', 'sonata.form.type.equal')
+            ->setPublic(true)
+            ->setDeprecated(
+                true,
+                'The "%alias_id%" service is deprecated since sonata-project/core-bundle 3.19 and will be removed in 4.0. Use "sonata.form.type.equal" instead.'
+            );
+    }
+
+    private function forceUserToMoveConfig(ContainerBuilder $container)
+    {
+        $bundles = $container->getParameter('kernel.bundles');
+
+        if (!isset($bundles['SonataFormBundle'])) {
+            return;
+        }
+
+        $defaultForm = [
+                'mapping' => [
+                    'enabled' => true,
+                    'type' => [],
+                    'extension' => [],
+                ],
+            ];
+        $defaultSerializer = [
+                'formats' => [
+                    0 => 'json',
+                    1 => 'xml',
+                    2 => 'yml',
+                ],
+            ];
+
+        if ($container->getParameter('sonata.core.form') !== $defaultForm) {
+            throw new \Exception('Move bundle config from sonata_core.form to sonata_form.form');
```

Comment:
Sure we should throw an exception? A deprecation message should be preferred for a smooth upgrade path.

Reply:
CoreBundle register own form and twig service based on core config when user do not register bridges.

When user register bridges then form and twig service will be loaded from bridges and bridges config, so config must be replace too. 

Register new bundles/bridges if equal to major change. Throw exception is awesome solution here becouse we can force user to do something. Mybe we should improve this  message?

For example:
"Your register form-extensions. From now registred SonataCoreBundle will be using form-extensions configuration. Please move sonata_core.form to sonata_form.form and keep it clear(empty)."

Reply:
> Throw exception is awesome solution here becouse we can force user to do something.

We should not *force* the user to do something when updating to a minor version

Reply:
And we don't. We force the user to do something in two cases:
1. Update CoreBundle and register new bridges from CoreBundle 
2. Update CoreBundle and register bridges from extensions 0.x

If user update bundles without register this bridges then we do not throw this exception.

Reply:
I missed this line:
https://github.com/sonata-project/SonataCoreBundle/pull/751/files#diff-2f96961c5d13c1ba7ad8ea345be1a57eR131